third_party/openocd/repos.bzl

```diff
@@ -0,0 +1,19 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+load("@bazel_tools//tools/build_defs/repo:git.bzl", "new_git_repository")
+
+def openocd_repos():
+    new_git_repository(
```

Comment:
Unfortunately, `git_repository` and `new_git_repository` are not compatible with airgapped builds since they have no `sha256` parameter, they are never cached in the repository cache. As we discussed offline, I think the easiest approach would be to try and use one or more `http_archive` rules (since you have submodule you need to init here), and if that doesn't work we can look at architecting something more substantial.

Also, as @drewmacrae pointed out, we should probably create a CI check to ensure there are no such repo rules used in our project. I created #15878 to track.

Reply:
Please use `http_archive_or_local` from `//rules/repo.bzl`.  Select an OpenOCD release from their releases page and reference the "source code" archive:
```
def openocd_repos(local = None):
    http_archive_or_local(
        name = "openocd"
        local = local,
        build_file = Label(...),
        sha256 = "...",
        strip_prefix = "<look in the tar file... something like openocd-0.11.1>",
        url = "https://github.com/...",
    )
```

The "or local" part of the rule allows a developer to modify the call to `openocd_repos` in the workspace file and specify a local path.  This is useful should the developer be debugging or enhancing the external dependency.

Reply:
I'm struggling with this because the OpenOCD releases don't include the code from git submodules. I can fetch these submodules independently with `http_archive_or_local`, but I'm not sure how to cobble together the src directory for consumption by `configure_make`.

For instance, OpenOCD assumes that the jimtcl code will be located at `jimtcl`, relative to the repo root. I figured out how to use `pkg_files` to describe the directory structure, but that may have been a false start because I can't seem to reify that into an actual directory — I can, unhelpfully, create a tar file.

I think I could make a custom rule that creates the exact symlinks I need, but I'm not sure if there's a more elegant way to do it.

Reply:
I see.  In that case, is there a pre-existing binary distribution that has what we want?

Reply:
The official distribution from sourceforge includes the dependencies (jimtcl, jaylink) that are expressed as git submodules in the github mirror.

I've sent a commit that uses sourceforge and changes the build to not have to bootstrap autotools.  Let me know if it works for you.